### PR TITLE
Sticky fixes and improvements.

### PR DESCRIPTION
**Pull request by titovmx**
- sticky was subscribed on theme change
- dom observe functions was placed to own service

### CODE REVIEW

File: src/ng/components/sticky/sticky.core.js [hunk 1]
```diff
@@ -44,13 +45,11 @@ class StickyCore extends Directive(STICKY_CORE_NAME, {view: `^^${VIEW_CORE_NAME}
...
 
-		sticky.invalidated.on(() => {
-			self.$scope.$apply();
-		});
+		this.view.theme.changed.on(() => self.$timeout(() => sticky.invalidate()));
 
-		this.$scope.$on('$destroy', () => {
-			sticky.destroy();
-		});
+		sticky.invalidated.on(() => self.$scope.$apply());
```
Comment (vkorolev) [inline src/ng/components/sticky/sticky.core.js]: Maybe `sticky.invalidate.on` ?

Comment (titovmx) [reply src/ng/components/sticky/sticky.core.js]: Sticky has already method ```invalidate```. So ```invalidated``` will be an event when invalidation finish.

Comment (klumba12) [reply src/ng/components/sticky/sticky.core.js]: +1

File: src/core/services/observe.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
+export function children(element, handler) {
```
Comment (klumba12) [inline src/core/services/observe.js]: maybe dom.observe is better name. it indicates that observations can occur only on dom elemeents

File: src/core/services/observe.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+	const observer = new MutationObserver((mutations) => {
+		mutations.forEach((mutation) => {
```
Comment (klumba12) [inline src/core/services/observe.js]: can remove parenthesis mutation =>

File: src/core/services/observe.js (hunk 2 above)
Comment (klumba12) [reply src/core/services/observe.js]: what do you think about next variants:
```javascript
 observe.children(element, e => ) //current
 observe.children(element).on(e =>
 observe(element).children.on(e => ...)
 observe(element).children(e =>
 observe(element).childrenChanged.on(e =>
```

**Review by klumba12 — APPROVED** [review]

**Pull request merged**